path_login.go

```diff
@@ -401,7 +407,33 @@ func (b *azureAuthBackend) verifyResource(ctx context.Context, subscriptionID, r
 	// Ensure the token OID is the principal id of the system-assigned identity
 	// or one of the user-assigned identities
 	if _, ok := principalIDs[claims.ObjectID]; !ok {
```

Comment:
Curious if we still require one of `vm_name`, `vmss_name`, or `resource_id` to be provided? Do you know if it still makes sense for that requirement to be true in the WIF environment?

Reply:
I believe you're right, we should check app_id if authentication vial vm/vmss/resource_id fails *or* if no such identifier was provided.

Reply:
i tweaked the switch branches a bit for this